src/include/aerospike/as_operations.h

```diff
@@ -776,35 +776,120 @@ as_operations_add_touch(as_operations* ops);
 AS_EXTERN bool
 as_operations_add_delete(as_operations* ops);
 
+// If you add flags in bits 4 or higher, make sure you make a corresponding
+// change in as_exp_path_modify_flags as well (and vice versa, of course).
+// Only bits 0-3 are unique to select vs. modify operations.
+/**
+ * These flags apply to the as_exp_select_by_path()/
+ * as_operations_select_by_path() functions.
+ */
+typedef enum {
+	/**
+	 * Return a tree from the root (bin) level to the bottom of the tree,
+	 * with only non-filtered out nodes.
+	 */
+	AS_EXP_PATH_SELECT_MATCHING_TREE = 0,
+
+	/**
+	 * Return the list of the values of the nodes finally selected by the context.
+	 * For maps, this returns the value of each (key, value) pair.
+	 */
+	AS_EXP_PATH_SELECT_VALUE = 1,
+
+	/**
+	 * Return the list of the values of the nodes finally selected by the context.
+	 * This is a synonym for AS_EXP_PATH_SELECT_VALUE to make it clear in your
+	 * source code that you're expecting a list.
+	 */
+	AS_EXP_PATH_SELECT_LIST_VALUE = 1,
+
+	/**
+	 * Return the list of map values of the nodes finally selected by the context.
+	 * This is a synonym for AS_EXP_PATH_SELECT_VALUE to make it clear in your
+	 * source code that you're expecting a map.  See also
+	 * AS_EXP_PATH_SELECT_MAP_KEY_VALUE.
+	 */
+	AS_EXP_PATH_SELECT_MAP_VALUE = 1,
+
+	/**
+	 * Return the list of map keys of the nodes finally selected by the context.
+	 */
+	AS_EXP_PATH_SELECT_MAP_KEY = 2,
+
+	/**
+	 * Returns the list of map (key, value) pairs of the nodes finally selected
+	 * by the context.  This is a synonym for setting both
+	 * AS_EXP_PATH_SELECT_MAP_KEY and AS_EXP_PATH_SELECT_MAP_VALUE bits together.
+	 */
+	AS_EXP_PATH_SELECT_MAP_KEY_VALUE = AS_EXP_PATH_SELECT_MAP_KEY | AS_EXP_PATH_SELECT_MAP_VALUE,
+
+	/**
+	 * If the expression in the context hits an invalid type (e.g., selects
+	 * as an integer when the value is a string), do not fail the operation;
+	 * just ignore those elements.  Interpret UNKNOWN as false instead.
```

Comment:
```suggestion
	 * just ignore those elements.  Interpret an expression that returns UNKNOWN as false instead.
```

Last sentence was a bit confusing for me.